agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -236,6 +253,54 @@ def calculate_container_time(container)
       0
     end
 
+    # @param [Array<Vmstat::Cpu>] prev_cpu_stats
+    # @param [Array<Vmstat::Cpu>] current_cpu_stats
+    # @return [Hash]
+    def calculate_average_cpu(prev_cpu_stats, current_cpu_stats)
+      all = prev_cpu_stats.zip(current_cpu_stats).map do |prev, current|
+        system_ticks = current.system - prev.system
+        user_ticks = current.user - prev.user
+        idle_ticks = current.idle - prev.idle
+
+        total_ticks = system_ticks + user_ticks + idle_ticks
+
+        [system_ticks, user_ticks, idle_ticks].map do |ticks|
+          (ticks / total_ticks.to_f) * 100.0
+        end
+      end
+
+      averages = all.transpose.map do |stats|
```

Comment:
BTW re summing up CPU utilization %, there's two schools of thought.. This kind of averaging is like Windows, where two busy-looping processes on a four-core machine will be 50% usage. Generally, in Linux (like what `top` etc do), two busy-looping processes/threads on an N-core machine will show as 200% usage.

I personally prefer the Linux style... You don't need to know how many CPUs your machine has to see if something is bottlenecking on a single CPU .